pkg/ingester/ingester.go

```diff
@@ -960,13 +991,17 @@ func (i *Ingester) PushWithCleanup(ctx context.Context, req *mimirpb.WriteReques
 	// retain anything from `req` past the exit from this function.
 	defer cleanUp()
 
-	// If we're using grpc handlers, we don't need to start/finish request here.
-	if !i.cfg.LimitInflightRequestsUsingGrpcMethodLimiter {
+	// Only start/finish request here, when the request comes not from grpc handlers (i.e. from ingest.Store).
+	// NOTE: when ingest storage's enabled, for a grpc request this will be a noop.
```

Comment:
I don't understand what this comment ("for a grpc request this will be a noop.")  is trying to tell.